library/common-library/templates/_hostnetwork.tpl

```diff
@@ -0,0 +1,34 @@
+{{- /*
+Abstraction of the hostNetwork toggle.
+This helper allows to override the global `.global.hostNetwork` with the value of `.hostNetwork`.
+Returns "true" if `hostNetwork` is enabled, otherwise fallbacks to a overridable function
+*/ -}}
+{{- define "common.hostNetwork" -}}
+{{- /* This allows us to use `$global` as an empty dict directly in case `Values.global` does not exists */ -}}
+{{- $global := index .Values "global" | default dict -}}
+
+{{- /* `get` will return "" (empty string) if value is not found, and the value otherwise, so we can type-assert with kindIs */ -}}
+{{- if get .Values "hostNetwork" | kindIs "bool" -}}
+    {{- if .Values.hostNetwork -}}
+        true
+    {{- else -}}
+        false
+    {{- end -}}
```

Comment:
Why not just
```suggestion
    {{- .Values.hostNetwork }}
```

In the end, we have already asserted it is bool so it does not make much sense to if-else on it.

Reply:
I left it when I did not assert that it is a bool. And I am still in doubt if I should fail when it is not a bool or not.

Solved in [#76799db](https://github.com/newrelic/helm-charts/pull/750/commits/76799db5ccad9eef227078fea726a82d6d12ea35)